app/com/github/pedrovgs/dotto/interpreter/DottoInterpreter.scala

```diff
@@ -0,0 +1,46 @@
+package com.github.pedrovgs.dotto.interpreter
+
+import akka.actor.ActorRef
+import cats.{Id, ~>}
+import com.github.pedrovgs.dotto.core.algebra.dotto._
+import com.github.pedrovgs.dotto.interpreter.actors.ShowMessageActor.ShowMessage
+import com.github.pedrovgs.dotto.interpreter.led.ledController._
+import com.github.pedrovgs.dotto.interpreter.morse.morseTranslator._
+import play.api.Logger
+
+/**
+  * Production interpreter implemented using Cats as core library.
+  */
+class DottoInterpreter(val showMessageActor: ActorRef) extends (Instruction ~> Id) {
+
+  def apply[A](i: Instruction[A]): Id[A] = i match {
+    case EnqueueMessage(messageToEnqueue) => enqueueMessage(messageToEnqueue)
```

Comment:
I see you point with redundancy and I think using Free in this context does not buy you much but if you had a ton more biz logic down the road you'd see the benefits.